apps/src/lab2/views/components/Instructions/ResourcePanel/SettingsPanel.tsx

```diff
@@ -2,27 +2,37 @@ import CloseButton from '@code-dot-org/component-library/closeButton';
 import {useTheme} from '@code-dot-org/component-library/common/contexts';
 import {SimpleDropdown} from '@code-dot-org/component-library/dropdown';
 import {Heading2} from '@code-dot-org/component-library/typography';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 
-import {LocaleProps} from '@cdo/apps/lab2/types';
 import {Setting} from '@cdo/apps/lab2/views/components/Instructions/ResourcePanel';
+import localization from '@cdo/apps/localization';
+import {LanguageInfo} from '@cdo/apps/localization/Localization';
 import {commonI18n} from '@cdo/apps/types/locale';
 
 import styles from './settings-panel.module.scss';
 interface SettingsPanelProps {
   settings: Setting[];
   closePanel: () => void;
-  localeProps?: LocaleProps;
 }
 
 const SettingsPanel: React.FunctionComponent<SettingsPanelProps> = ({
   settings,
   closePanel,
-  localeProps,
 }) => {
   const {theme} = useTheme();
   // SimpleDropdown isn't themed properly, so we have to manually set the color.
   const dropdownColor = theme === 'Dark' ? 'white' : 'black';
+  const [locale, setLocale] = useState<string>(localization.locale);
+  const [localeOptions, setLocaleOptions] = useState<LanguageInfo[]>(
+    localization.locales
+  );
+
+  useEffect(() => {
```

Comment:
Instead of using state to hold the locale, the `useLocalization` hook already does that. So you can just do: 

```
const locale = useLocalization();

useEffect(() => setLocaleOptions(localization.locales), [locale]);
```

though if this pattern is more prevalent I could consider adding the ability to pass in a callback to the hook that would run on a locale change.

Reply:
updated! It might be nice to make the hook more robust so we can get other values out of the change event. Also, should the hook ever remove listeners (as a return from the `useEffect`, for example)? I couldn't find a way to remove listeners from `localization`.